[VDG] Dual Currency Input Minor layout improvement

ichthus1604 opened the pull request:
Clicking anywhere in the right side of the dual currency entry box now properly focuses the TextPresenter

yahiheb reviewed (approved):
tACK

MaxHillebrand commented:
What do you mean with caret?

The "is roughly"  wavy equal symbol should never be on the Bitcoin amount.

soosr commented:
> 



> What do you mean with caret?
> 
> The "is roughly" wavy equal symbol should never be on the Bitcoin amount.

The blinking `|` sign that marks you what position you will type at when you click on a TextBox.

ichthus1604 commented:
@soosr can't reproduce, I can see caret properly on both dark and light themes:

![image](https://user-images.githubusercontent.com/98904713/163674934-d2f7e0f4-34a2-4c1d-a9cf-d918416c436f.png)

![image](https://user-images.githubusercontent.com/98904713/163674951-010dfde2-616e-4a47-a532-4c4a0f99d56d.png)

Also tried changing window size, etc. Still couldn't reproduce.

yahiheb commented:
I can't reproduce it either.

danwalmsley commented on WalletWasabi.Fluent/Controls/DualCurrencyEntryBox.axaml.cs:
```diff
@@ -372,15 +372,20 @@ private void ReorganizeVisuals()
 			{
-				_leftEntryBox.HorizontalAlignment = Avalonia.Layout.HorizontalAlignment.Right;
-				_rightEntryBox.HorizontalAlignment = Avalonia.Layout.HorizontalAlignment.Stretch;
+				_leftEntryBox.TextAlignment = Avalonia.Media.TextAlignment.Right;
```
Add a psuedo class: :isbtc :isfiat 

and then set those in a style like : `Selector="DualCurrencyEntryBox:isFiat" Setter="Property ="HorizontalContentAligntment="x"

danwalmsley replied:
`PseudoClasses.Set(":expanded", value);`

soosr commented:
I don't do anything special. I am not even sure how could I debug it...

![caretissue](https://user-images.githubusercontent.com/16364053/163973787-f38f3d91-4768-4dea-bb22-b40a56a98506.gif)

ichthus1604 commented:
@soosr I have made a good looking improvement to the caret:

![DualInputCaretOffset](https://user-images.githubusercontent.com/98904713/164362826-40fae141-7516-4646-bce0-f7dbac11f296.gif)

Caret will not appear in the correct spot when inputs are empty, so as to fit nicely with the watermarks below.

@danwalmsley I've removed the ugly property manipulation in code, and replaced it by pseudoclasses. However, I've kept the code that manually sets `Grid.Column` because if I remove that, tab key focus navigation breaks.

soosr reviewed (commented):
The left side currency unit should be next to the amount.

![image](https://user-images.githubusercontent.com/16364053/164410784-6d634a58-fd61-4be2-87fd-a02181bd4b3b.png)

ichthus1604 commented:
@soosr 

> The left side currency unit should be next to the amount.

fixed.

yahiheb reviewed (approved):
tACK

soosr reviewed (approved):
tACK
